### PR TITLE
[apps] Add minimal app with dev-client

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Added a bare-bone app that includes dev-client. 

# How

- run `expo init` 
- added dev-client plugin
- run `expo eject` 
- removed all unused dependencies (expo-updates was removed too - will be added in the future)
- used a new liking module 

# Test Plan

- Android & iOS ✅

### CODE REVIEW

File: apps/dev-client/android/build.gradle [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// Top-level build file where you can add configuration options common to all sub-projects/modules.
+
+buildscript {
+    ext {
+        buildToolsVersion = "29.0.3"
```
Comment (fson) [inline apps/dev-client/android/build.gradle:5]: In the template this had to be updated to `buildToolsVersion = "30.0.3"` for the Android build to work – does it need to be updated here as well?  Or is that only when using the upstream React Native version?

File: apps/dev-client/ios/Podfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+platform :ios, '11.0'
+
+target 'devclient' do
+  use_expo_modules!
+
+  config = use_native_modules!
+
+  pod 'expo-dev-menu', path: '../../../packages/expo-dev-menu', :configurations => :debug
+  pod 'expo-dev-menu-interface', path: '../../../packages/expo-dev-menu-interface', :configurations => :debug
```
Comment (fson) [inline apps/dev-client/ios/Podfile:13]: IIRC this pod needed to be included in the release builds too?

File: apps/dev-client/package.json [hunk 3]
```diff
@@ -0,0 +1,116 @@
+{
+  "name": "dev-client",
+  "version": "0.0.1",
+  "main": "index.js",
+  "scripts": {
+    "postinstall": "expo-yarn-workspaces postinstall",
+    "android": "react-native run-android",
+    "ios": "react-native run-ios",
```
Comment (fson) [inline apps/dev-client/package.json:8]:
```suggestion
    "android": "expo run:android",
    "ios": "expo run:ios",
```
Maybe?

**Review by fson — APPROVED** [review]
🚀

**Review by esamelson — CHANGES_REQUESTED** [review]
Could we rename this to something else (maybe `bare-sandbox` since it's sort of an analog of the existing `sandbox` app)? In the past there has been pushback to adding more apps to this repo's workspace, to avoid cluttering it with mostly unused apps that must be kept up to date (like we used to have in universe). IMO there is justification for adding this app in part because it's useful when developing many different modules, not just the dev client, so it would be great to have the name communicate as such! 🙂 

And to add to the test plan, could you confirm/verify (maybe by doing a dry run) that `et publish` updates dependencies in this app automatically, like it does with NCL/bare-expo/etc?

**Comment by lukmccall** [comment]
> And to add to the test plan, could you confirm/verify (maybe by doing a dry run) that `et publish` updates dependencies in this app automatically, like it does with NCL/bare-expo/etc?

Project dependencies will be updated by `et publish`. This script looks at the output of `yarn workspaces info`. If a project is on that list, it will update dependencies. All projects from the apps folder are included in our workspace.

**Review by esamelson — APPROVED** [review]

**Pull request merged**